### PR TITLE
making global search

### PR DESCRIPTION
**Pull request by kristofferj**

### CODE REVIEW

File: src/globalsearch/GlobalSearch.js [hunk 1]
```diff
@@ -0,0 +1,230 @@
...
+
+  handleInputChange = event => {
+    const query = event.target.value
+    this.setState({
+      inputValue: query
+    })
+    this.props.onSearch(query)
+  }
+
+  handleItemClick = item => {
```
Comment (bjoerge) [inline src/globalsearch/GlobalSearch.js]: It think we can do with just `handleItemClick` as it just calls `this.handleChange` that only calls `this.pops.onChange`, right?

File: src/globalsearch/GlobalSearch.js [hunk 2]
```diff
@@ -0,0 +1,230 @@
...
+        </div>
+        {
+          ((topItems && topItems.length > 0) || (items && items.length > 0)) && (
+            <div className={isOpen ? styles.listContainer : styles.listContainerClosed}>
+              <ul className={styles.items}>
+                {
+                  topItems && topItems.length > 0 && topItems.map((item, i) => {
+                    return (
+                      <li
+                        onClick={this.handleItemClick.bind(null, item)}
```
Comment (bjoerge) [inline src/globalsearch/GlobalSearch.js]: Do the onClick have to be triggered on the `<li>`, or could click handling be the responsibility of the whoever implements `renderItem`?

File: src/globalsearch/GlobalSearch.js [hunk 3]
```diff
@@ -0,0 +1,230 @@
+import React, {PropTypes} from 'react'
+import styles from 'part:@sanity/components/globalsearch/default-style'
+import SearchIcon from 'part:@sanity/base/search-icon'
+import Spinner from 'part:@sanity/components/loading/spinner'
+import enhanceWithClickOutside from 'react-click-outside'
+
+class GlobalSearch extends React.Component {
+  static propTypes = {
```
Comment (bjoerge) [inline src/globalsearch/GlobalSearch.js]: Are any of these required? Maybe add `.isRequired` to the ones that are.

File: src/globalsearch/GlobalSearch.js (hunk 2 above)
Comment (kristofferj) [inline src/globalsearch/GlobalSearch.js]: The keyboard-navigation is supposed to do the same on press "Enter", so i find it more consistent to call the same function for both click and press "Enter".

**Pull request merged**